mkosi/qemu.py

```diff
@@ -384,8 +381,11 @@ def start_virtiofsd(
         # as root without the source directories necessarily being owned by root. We achieve this by running
         # virtiofsd as the owner of the source directory and then mapping that uid to root.
 
+        if name is None:
+            name = f"{config.machine_or_name()}-{systemd_escape(config, directory, path=True)}"
```

Comment:
We should probably also escape the name to be on the safe side and falling back to the default name for an empty name would probably also make sense
```suggestion
        if not name:
            name = f"{config.machine_or_name()}-{systemd_escape(config, directory, path=True)}"
        else:
            name = systemd_escape(config, name)
```